pkg/core/resources/apis/meshservice/status/updater.go

```diff
@@ -176,32 +176,21 @@ func buildDataplaneProxies(
 			}
 		}
 		healthyInbounds := 0
-		for _, port := range ports {
-			if inbound := dpInboundForMeshServicePort(dpp.Spec.GetNetworking().Inbound, port); inbound != nil {
-				if inbound.State == mesh_proto.Dataplane_Networking_Inbound_Ready {
+		for _, meshServicePort := range meshServicePorts {
+			for _, dpInbound := range dpp.Spec.GetNetworking().Inbound {
+				if dpInbound.State == mesh_proto.Dataplane_Networking_Inbound_Ready &&
+					meshservice.MatchInboundWithMeshServicePort(dpInbound, meshServicePort) {
```

Comment:
once at least one inbound is matched you can stop iterating over the inbounds and move on to the next port

Reply:
Good point